src/Kinesis.php

```diff
@@ -0,0 +1,103 @@
+<?php
+
+namespace PodPoint\MonologKinesis;
+
+use Aws\Kinesis\KinesisClient;
+use Illuminate\Config\Repository;
+use Illuminate\Support\Arr;
+use PodPoint\MonologKinesis\Contracts\Client;
+
+class Kinesis implements Client
+{
+    /** @var KinesisClient */
+    protected $kinesis;
+
+    /** @var Repository */
+    protected $config;
+
+    public function __construct(Repository $config)
+    {
+        $this->config = $config;
+    }
+
+    /**
+     * Apply configuration to this client.
+     *
+     * @param  array  $channelConfig
+     * @return $this
+     */
+    public function configure(array $channelConfig): Kinesis
+    {
+        $this->kinesis = $this->configureKinesisClient($channelConfig);
```

Comment:
Can this be done in the constructor?

Reply:
Okay so if you just want to have
```php
$client = $app->make(Client::class);
```
then you need to use something like
```php
$this->app->bind(Client::class, Kinesis::class);
// ...
$manager->extend('kinesis', function (Container $app, array $config) {
    $app->when(Client::class)->needs('$channelConfig')->give($config);
    // ...
});
```
which in my opinion start to be a bit messy around the IoC so I'd rather keep the configure method if that's okay with you...